src/oscar/apps/offer/benefits.py

```diff
@@ -203,6 +204,50 @@ def apply(
         return BasketDiscount(discount)
 
 
+class AbsoluteProductDiscountBenefit(AbsoluteDiscountBenefit):
+    """
+    An offer benefit that gives an absolute discount on each applicable product.
+    The value for this benefit refers to the maximum discount per product
+    regardless of the quantity.
+    """
+
+    class Meta:
+        app_label = "offer"
+        proxy = True
+        verbose_name = _("Absolute product discount benefit")
+        verbose_name_plural = _("Absolute product discount benefits")
+
+    def get_lines_to_discount(self, offer, line_tuples):
```

Comment:
Is there no default implementation of ``get_lines_to_discount`` on Benefit that works appropriately? Is this implementation really different from other discounts?

Reply:
There is no default implementation of `get_lines_to_discount`. Each benefit uses slightly different version of this code.